public/js/graph/react_graph.js

```diff
@@ -438,6 +440,34 @@ var Graph = React.createClass({
         });
     },
 
+    onButtonPress: function(zoomOrPanFunction, direction) {
+        this.setState({mouseDown: setInterval(() => zoomOrPanFunction(direction), 100)});
```

Comment:
I'm about to update the pull request. I think I fixed the hack that you mentioned, but I just wanted to confirm that I understood the problem correctly (and fixed it correctly). 

Is it the case that the old code (in the snippet above) sets the state over and over at each interval? If so, my solution (included in my update) is to use a variable to get the return value from `setInterval` and then just pass the variable to set the state of `mouseDown`. 

If this isn't the correct approach, then my feeling is that maybe I need to find a different way to pass the return value from `setInterval` to `clearInterval` rather than using state to do this.

Also, now controlling the graph with the buttons is quite slow (by changing the interval to 500) and using the mouse wheel and arrow buttons are quite fast. Should I try to find a way to slow down the mouse wheel zoom and the arrow button panning?